### PR TITLE
Update sample script to strip colons from cert fingerprint

### PR DESCRIPTION
**Pull request by danewalton**

### CODE REVIEW

**Comment by danewalton** [comment]
@CIPop I remembered the label this time :)

**Comment by hihigupt** [comment]
> openssl ecparam -out device_ec_key.pem -name prime256v1 -genkey

set -o errexit # Exit if command failed.
set -o nounset # Exit if variable not set.
set -o pipefail # Exit if pipe failed.

can we set this in script?

---
Refers to: sdk/iot/hub/samples/src/generate_certificate.sh:5 in edd78b0. [](commit_id = edd78b0d27d459b4f7e7a74145b53da82baa47fe, deletion_comment = False)

File: sdk/iot/hub/samples/src/generate_certificate.sh [hunk 1]
```diff
@@ -22,5 +22,8 @@ echo -e "\nSample certificate generated"
 echo "Use the device_cert_store.pem file within the sample:"
 echo    export AZ_IOT_DEVICE_X509_CERT_PEM_FILE=$(pwd)/device_cert_store.pem
 
-echo -e "\nUse the following fingerprint when creating your device in IoT Hub (must remove colons):"
-openssl x509 -noout -fingerprint -in device_ec_cert.pem
+echo -e "\nUse the following fingerprint when creating your device in IoT Hub:"
+FINGERPRINT=`openssl x509 -noout -fingerprint -in device_ec_cert.pem`
```
Comment (hihigupt) [inline sdk/iot/hub/samples/src/generate_certificate.sh:26]: >openssl x509 -noout -fingerprint -in device_ec_cert.pem [](start = 13, length = 55)

openssl x509 -noout -fingerprint -in device_ec_cert.pem | tee | sed 's/://g' > fingerprint.txt

**Comment by hihigupt** [comment]
> openssl req -new -days 365 -nodes -x509 -key device_ec_key.pem -out device_ec_cert.pem -config x509_config.cfg -subj "/CN=paho-sample-device1"

how about taking this as argument ? But by default set it to paho-sample-device1

CN_NAME=${1:-"paho-sample-device1"} 

openssl req -new -days 365 -nodes -x509 -key device_ec_key.pem -out device_ec_cert.pem -config x509_config.cfg -subj "/CN=$CN_NAME"

---
Refers to: sdk/iot/hub/samples/src/generate_certificate.sh:6 in edd78b0. [](commit_id = edd78b0d27d459b4f7e7a74145b53da82baa47fe, deletion_comment = False)

File: sdk/iot/hub/samples/src/generate_certificate.sh [hunk 2]
```diff
@@ -22,5 +22,8 @@ echo -e "\nSample certificate generated"
 echo "Use the device_cert_store.pem file within the sample:"
 echo    export AZ_IOT_DEVICE_X509_CERT_PEM_FILE=$(pwd)/device_cert_store.pem
```
Comment (hihigupt) [inline sdk/iot/hub/samples/src/generate_certificate.sh:23]: >echo [](start = 0, length = 4)

ask them to run this command at their terminal, if they want to set the environment variable

File: sdk/iot/hub/samples/src/generate_certificate.sh (hunk 1 above)
Comment (hihigupt) [reply sdk/iot/hub/samples/src/generate_certificate.sh:26]: In windows you can open these cert by double click and get the hash. (by just renaming pem -> crt)

---
In reply to: [421902872](https://github.com/Azure/azure-sdk-for-c/pull/702#discussion_r421902872) [](ancestors = 421902872)

**Comment by danewalton** [comment]
> 
> 
> > openssl req -new -days 365 -nodes -x509 -key device_ec_key.pem -out device_ec_cert.pem -config x509_config.cfg -subj "/CN=paho-sample-device1"
> 
> how about taking this as argument ? But by default set it to paho-sample-device1
> 
> CN_NAME=${1:-"paho-sample-device1"}
> 
> openssl req -new -days 365 -nodes -x509 -key device_ec_key.pem -out device_ec_cert.pem -config x509_config.cfg -subj "/CN=$CN_NAME"
> 
> Refers to: sdk/iot/hub/samples/src/generate_certificate.sh:6 in [edd78b0](https://github.com/Azure/azure-sdk-for-c/commit/edd78b0d27d459b4f7e7a74145b53da82baa47fe). [](commit_id = [edd78b0](https://github.com/Azure/azure-sdk-for-c/commit/edd78b0d27d459b4f7e7a74145b53da82baa47fe), deletion_comment = False)

It's not entirely relevant. The CN needs to match in the CA case but not self signed

File: sdk/iot/hub/samples/src/generate_certificate.sh (hunk 2 above)
Comment (danewalton) [reply sdk/iot/hub/samples/src/generate_certificate.sh:23]: So we should not be setting these?

File: sdk/iot/hub/samples/src/generate_certificate.sh (hunk 1 above)
Comment (danewalton) [reply sdk/iot/hub/samples/src/generate_certificate.sh:26]: To comment one: Is that just another way of doing it? Does it add a benefit?
To two: Yes this just functions as a catch all solution

**Comment by danewalton** [comment]
> 
> 
> > openssl ecparam -out device_ec_key.pem -name prime256v1 -genkey
> 
> set -o errexit # Exit if command failed.
> set -o nounset # Exit if variable not set.
> set -o pipefail # Exit if pipe failed.
> 
> can we set this in script?
> 
> Refers to: sdk/iot/hub/samples/src/generate_certificate.sh:5 in [edd78b0](https://github.com/Azure/azure-sdk-for-c/commit/edd78b0d27d459b4f7e7a74145b53da82baa47fe). [](commit_id = [edd78b0](https://github.com/Azure/azure-sdk-for-c/commit/edd78b0d27d459b4f7e7a74145b53da82baa47fe), deletion_comment = False)

How about those but without the errexit one. If the rm command fails it terminates (in the case that the file isn't there).

File: sdk/iot/hub/samples/src/generate_certificate.sh (hunk 1 above)
Comment (hihigupt) [reply sdk/iot/hub/samples/src/generate_certificate.sh:26]: yes, simpler and no need to create variables.

File: sdk/iot/hub/samples/src/generate_certificate.sh (hunk 2 above)
Comment (hihigupt) [reply sdk/iot/hub/samples/src/generate_certificate.sh:23]: well you can not set these variable for them, but you can print command that they should execute in the shell

File: sdk/iot/hub/samples/src/generate_certificate.sh (hunk 1 above)
Comment (danewalton) [reply sdk/iot/hub/samples/src/generate_certificate.sh:26]: Gotcha good to know thanks. Updated

File: sdk/iot/hub/samples/src/generate_certificate.sh (hunk 2 above)
Comment (danewalton) [reply sdk/iot/hub/samples/src/generate_certificate.sh:25]: gotcha. updated

File: sdk/iot/hub/samples/src/generate_certificate.sh [hunk 3]
```diff
@@ -2,11 +2,13 @@
...
 
+set -o nounset # Exit if variable not set.
+set -o pipefail # Exit if pipe failed.
+
 openssl ecparam -out device_ec_key.pem -name prime256v1 -genkey
 openssl req -new -days 365 -nodes -x509 -key device_ec_key.pem -out device_ec_cert.pem -config x509_config.cfg -subj "/CN=paho-sample-device1"
 openssl x509 -noout -text -in device_ec_cert.pem
 
-
 rm device_cert_store.pem
```
Comment (hihigupt) [inline sdk/iot/hub/samples/src/generate_certificate.sh:12]: you could use rm -f device_cert_store.pem

**Comment by hihigupt** [comment]
> openssl ecparam -out device_ec_key.pem -name prime256v1 -genkey

command failure is important to exit script, for rm use -f that will pass this check.

---
In reply to: [625595344](https://github.com/Azure/azure-sdk-for-c/pull/702#issuecomment-625595344) [](ancestors = 625595344)

---
Refers to: sdk/iot/hub/samples/src/generate_certificate.sh:5 in edd78b0. [](commit_id = edd78b0d27d459b4f7e7a74145b53da82baa47fe, deletion_comment = False)

File: sdk/iot/hub/samples/src/generate_certificate.sh (hunk 3 above)
Comment (danewalton) [reply sdk/iot/hub/samples/src/generate_certificate.sh:12]: done

**Pull request merged**